src/diagonal.works/b6/api/api.go

```diff
@@ -159,19 +177,26 @@ func Less(a interface{}, b interface{}) (bool, error) {
 			return a.Less(b), nil
 		}
 	}
+
 	return false, fmt.Errorf("can't compare %T with %T", a, b)
 }
 
 func Equal(a interface{}, b interface{}) (bool, error) {
-	switch a := a.(type) {
-	case int:
-		if b, ok := b.(int); ok {
-			return a == b, nil
+	if ConvertibleToFloat64(a) && ConvertibleToFloat64(b) {
```

Comment:
I'm perhaps being picky here, but this changes the semantics of equal, since converting values from ints to floats can result in things that were not equal becoming equal (eg an integer that's too large to fit in float64 without being rounded, and the rounded float64 value). I think we should be a little picky since the function is called "Equal". I think it's fine to have another function, like EqualFloat64 or similar that does this though.